core/src/main/scala/spinoco/fs2/cassandra/internal/CodecWriter.scala

```diff
@@ -0,0 +1,51 @@
+package spinoco.fs2.cassandra.internal
+
+import com.datastax.oss.driver.api.core.ProtocolVersion
+import com.datastax.oss.driver.api.core.data.{SettableByIndex, SettableByName}
+import spinoco.fs2.cassandra.CType
+import spinoco.fs2.cassandra.internal.CodecSerializer.CodecSerializeSyntax
+
+import java.nio.ByteBuffer
+
+object CodecWriter {
+  implicit class CodecWriteSyntax[V](val tpe: CType[V]) extends AnyVal {
+
+    def writeByNameSerialized[D <: SettableByName[D]](key: String, value: V, data: D, protocolVersion: ProtocolVersion): D = {
+      tpe
+        .serialize(value, protocolVersion)
+        .toEither
+        .left.map { e => throw new Throwable(e.message) }
+        .right.map(bv => data.setBytesUnsafe(key, bv))
+        .getOrElse(data)
```

Comment:
I am not sure if I follow this here -> left creates and exception, and then, we do getOrElse to return original data, meaning exception gets ignored ? We shall surelly throw here if we cannot resovle in different way

Reply:
Actually, doesn't this throw the exception? I mean, there surely is a better idiom to do write this (I am going to refactor this shortly), but it seems to me like this does throw the exception.

Reply:
(
Unless you have another preference, I am going to use fold for this idiom:
```
        .fold(
e => throw new Throwable(e.message),
bv => data.setBytesUnsafe(key, bv)
)
```
)

Reply:
tpe
        .serialize(value, protocolVersion)
        .map(data.setBytesUnsafe(idx, _))
        .getOrElse(throw new Throwable("Failed to serialize"))

Reply:
(Just for the record: above solution throws the message away, so leaving current code in place for the time being.)